### PR TITLE
Refactor the additional argument passing to dataset classes

### PR DESCRIPTION
**Pull request by stefdoerr**
This cleans up the logic of passing extra datasets arguments to dataset classes.

### CODE REVIEW

File: torchmdnet/datasets/md17.py [hunk 1]
```diff
@@ -25,16 +25,16 @@ class MD17(InMemoryDataset):
 
     available_molecules = list(molecule_files.keys())
 
-    def __init__(self, root, transform=None, pre_transform=None, dataset_arg=None):
-        assert dataset_arg is not None, (
+    def __init__(self, root, transform=None, pre_transform=None, molecules=None):
+        assert molecules is not None, (
```
Comment (raimis) [inline torchmdnet/datasets/md17.py:29]: Maybe it is better to check:
```python
assert molecules in MD17.available_molecules or molecules == "all"
```

File: torchmdnet/datasets/qm9.py [hunk 2]
```diff
@@ -5,14 +5,14 @@
 
 
 class QM9(QM9_geometric):
-    def __init__(self, root, transform=None, dataset_arg=None):
-        assert dataset_arg is not None, (
+    def __init__(self, root, transform=None, label=None):
+        assert label is not None, (
```
Comment (raimis) [inline torchmdnet/datasets/qm9.py:9]: Maybe it is better to check:
```python
assert label in qm9_target_dict.values()
```

**Review by raimis — COMMENTED** [review]
This look good in general. Just some argument checks can be more strict as commented.

File: torchmdnet/datasets/md17.py (hunk 1 above)
Comment (stefdoerr) [reply torchmdnet/datasets/md17.py:29]: Done, in a bit different way

File: torchmdnet/datasets/qm9.py (hunk 2 above)
Comment (stefdoerr) [reply torchmdnet/datasets/qm9.py]: Done

**Review by raimis — APPROVED** [review]

**Pull request merged**